Source/Basic-Interaction-Component/Runtime/Behaviors/UnsnapBehavior.cs

```diff
@@ -24,30 +24,37 @@ public class EntityData : IBehaviorData
         {
             [DataMember]
             [DisplayName("Object to unsnap")]
-            public ScenePropertyReference<ISnappableProperty> SnappedObject { get; set; }
-            
+            public SingleScenePropertyReference<ISnappableProperty> SnappedObject { get; set; }
+
             [DataMember]
             [DisplayName("Snap zone to unsnap")]
-            public ScenePropertyReference<ISnapZoneProperty> SnapZone { get; set; }
+            public SingleScenePropertyReference<ISnapZoneProperty> SnapZone { get; set; }
 
             public Metadata Metadata { get; set; }
-            
+
             /// <inheritdoc/>            
             [IgnoreDataMember]
             public string Name
             {
                 get
                 {
-                    string snappedObject = "[NULL]";
-                    string snapZone = "[NULL]";
+                    try
+                    {
+                        string snappedObject = "[NULL]";
+                        string snapZone = "[NULL]";
+
+                        if (SnappedObject.IsEmpty() == false || SnapZone.IsEmpty() == false)
+                        {
+                            snappedObject = SnappedObject.IsEmpty() ? "snapped object" : SnappedObject.Value.SceneObject.GameObject.name;
+                            snapZone = SnapZone.IsEmpty() ? "its snap zone" : SnapZone.Value.SceneObject.GameObject.name;
+                        }
 
-                    if(SnappedObject.IsEmpty() == false || SnapZone.IsEmpty() == false)
+                        return $"Unsnap {snappedObject} from {snapZone}";
+                    }
+                    catch
                     {
-                        snappedObject = SnappedObject.IsEmpty() ? "snapped object" : SnappedObject.Value.SceneObject.GameObject.name;
-                        snapZone = SnapZone.IsEmpty() ? "its snap zone" : SnapZone.Value.SceneObject.GameObject.name;
+                        return "Unsnap object";
```

Comment:
All object references have the same issue and that needs to be fixed. That said, I decided to enclose all name functions in try/catch blocks as a safety net, since an exception here would cause the entire drawer to crash miserably. Thing is, you should never see the text in the catch block, and now you do.